fairscale/optim/grad_scaler.py

```diff
@@ -38,36 +75,306 @@ def __init__(
         growth_interval: int = 2000,
         enabled: bool = True,
         process_group: Any = dist.group.WORLD,
-    ) -> None:
+    ):
         super().__init__(
             init_scale=init_scale,
             growth_factor=growth_factor,
             backoff_factor=backoff_factor,
             growth_interval=growth_interval,
             enabled=enabled,
         )
-        self.display_warning = True
-        self.group = process_group
+        if enabled and amp_definitely_not_available():  # type: ignore
```

Comment:
how does this work for the CPU only version of GradScaler? even though this is a CPU only version we are assuming that we have GPUs to run the model. just wondering if that is a fair assumption to make.

Reply:
I don't have a clear answer to this. Would the cpu-tests be enough to ensure that this works?